### ISSUE TITLE
hcloud-cloud-controller-manager isn't propagated to workers

### ISSUE THREAD

**Issue by voron** [xetys/hetzner-kube#160]
I just bootstraped k8s from hetzner-kube `0.3.2-SNAPSHOT`, and noted `hcloud-cloud-controller-manager` is not deployed on workers, neither during bootstrap nor during `cluster add-worker`. I had to follow https://github.com/xetys/hetzner-kube/issues/4#issuecomment-361984088 to get Hetzner metadata and IP to pull up into k8s.

**Comment by voron** [comment]
I checked my deploy again, and it looks like pre-created workers have `20-hcloud.conf`, but I need to re-add worker node to k8s to have Hetzner metadata such as ProviderID, region, zone to appear. `cluster add-worker` workers  don't have `20-hcloud.conf` at all;

**Comment by xetys** [comment]
Yes, that is desired to be this way. If I deploy the controller manager silently, you cannot use the `add extrernal worker` feature, as the cloud controller manager doesn't find the external server in the cloud (obviously) and deletes it immediately after registration. So I decided to move it to an add-on. So you can post deploy it using `addon install hcloud-cloud-controller-manager`

**Comment by voron** [comment]
In my case I bootstraped k8, installed addon `hcloud-cloud-controller-manager` and then added 2 workers via `cluster add-worker`. Should these auto-created by hetzner-kube workers have `20-hcloud.conf` as Hetzner cloud instances ?

**Comment by xetys** [comment]
Possibly we need to add some kind of hooks into the addon section. So we define several hooks for different steps like adding nodes, where add-ons can hook into to run some custom code.
